ui/cypress/integration/common/volume.js

```diff
@@ -68,11 +70,42 @@ Then(
       },
     );
 
+    // Check if the right side panel updates
+    cy.get('[data-cy="volume_detail_card_name"]').should(
+      'contain',
+      volumeNameSparseLoopDevice,
+    );
     // Check if all the labels are presents
-    cy.get('[data-cy=volume_label_name]').should('contain', volume_label_name);
-    cy.get('[data-cy=volume_label_value]').should(
+    cy.get('[data-cy="volume_label_name"]').should(
+      'contain',
+      volume_label_name,
+    );
+    cy.get('[data-cy="volume_label_value"]').should(
       'contain',
       volume_label_value,
     );
   },
 );
+
+And(
+  'I delete the volume by click on the delete volume button and confirm the deletion',
+  () => {
+    cy.get('[data-cy="delete_volume_button"]').click({ force: true });
+    cy.get('[data-cy="confirm_deletion_button"]').click();
+  },
+);
+
+Then('I check the volume disappear from the volume list', () => {
```

Comment:
You would be better off passing a specific volume name to all your steps, and checking it's not part of the list anymore. This check is too error-prone (only works when stars are aligned).

Reply:
Yes. I agree. 
I didn't figure out a way to check that this row is not part of the list anymore. Maybe we need to traverse the returned JQuery object here.
```
  cy.waitUntil(
    () =>
      cy.get('[data-cy="volume_table_row"]').then(($el) => $el.length === 3),
    {
      errorMsg: `Volume test-volume-sparse is not deleted`,
      timeout: 120000, // waits up to 120000 ms, default to 5000
      interval: 5000, // performs the check every 5000 ms, default to 200
    },
  );
```